### DISCUSSION TITLE
How to open "a layout" at VimEnter? (namely Neo-tree and Aerial)

### DISCUSSION THREAD

**Discussion by andreacfromtheapp** [LazyVim/LazyVim#3139] [Q&A] [answered]
Hi all,

I would like to open `neo-tree` and `aerial (symbols)` whenever a file is opened. I tried using `autocmds.lua` as [suggested on neo-tree discussion](https://github.com/nvim-neo-tree/neo-tree.nvim/discussions/679) to no avail. Ideally refined to selected specific file extensions.

```lua
-- Autocmds are automatically loaded on the VeryLazy event
-- Default autocmds that are always set: https://github.com/LazyVim/LazyVim/blob/main/lua/lazyvim/config/autocmds.lua
-- Add any additional autocmds here

local function augroup(name)
  return vim.api.nvim_create_augroup("lazyvim_" .. name, { clear = true })
end

-- open neo-tree at launch
vim.api.nvim_create_autocmd({ "VimEnter" }, {
  pattern = "*",
  group = augroup "neotreeonopen",
  once = true,
  callback = function(_)
    if vim.fn.argc() == 0 then
      vim.cmd "Neotree"
    end
  end,
})
```

Please and thank you.

ps shower thought: perhaps there is a LazyVim toggle that one can set in configs?

**Comment by dpetka2001** [comment]
That autocmd only opens neo-tree when you invoke Neovim without any file arguments, like when doing `nvim`. What is it exactly you're trying to achieve? How will you be opening the files?

**Reply by andreacfromtheapp** [reply]
Hi @dpetka2001,

interesting observation. I usually open nvim with either of the following:

- `nvim` (followed by invoking either `Projects` or `Restore Session` or `Recents Files`)
- with a file directly `nvim filepath`

I would need to have the desired "layout" opening (neotree and aerial) in both cases (the former after the dashboard action) whenever a file is opened.

As of today I usually use my `edgy layout` (`leader ue`) or manually open neo-tree (`leader e`) and aerial (`leader cs`) after I have opened a file with the above mentioned cases.

Please and thank you

**Reply by dpetka2001** [reply]
Try out the following
```lua
vim.api.nvim_create_autocmd({ "SessionLoadPost", "UIEnter" }, {
  pattern = { "*.lua", "*.py" },
  group = vim.api.nvim_create_augroup("UIEnter", { clear = true }),
  desc = "Open neo-tree on UIEnter",
  callback = function()
    vim.cmd("Neotree")
  end,
})
```
There might be some edge cases, in that case you should read `:h events` for maybe choosing more appropriate events according to your needs.

**Reply by andreacfromtheapp** [reply]
thanks :) really appreciated!

it does kind of work except that:

- neotree path is wherever nvim was invoked and not the file path
- I tried adding `AerialOpen`
- I tried adding CTRL+W to it to move the cursor to the file

Adding extra commands breaks it.

Now that I think of it, is there a way to make it invoke Edgy (`leader ue`) instead? That way one would customize the layout in `edgy.lua` and the `autocmd` would be agnostic.

Please and thank you

**Reply by dpetka2001** [reply, marked as answer]
For me the path always opens where the file is. 

For `edgy` and moving the cursor to file you could do
```lua
vim.api.nvim_create_autocmd({ "SessionLoadPost", "UIEnter" }, {
  pattern = { "*.lua", "*.py" },
  group = vim.api.nvim_create_augroup("UIEnter", { clear = true }),
  desc = "Open neo-tree on UIEnter",
  callback = function()
    -- vim.cmd("Neotree")
    require("edgy").toggle()
    vim.schedule(function()
      vim.cmd("wincmd l")    -- lowercase L
    end)
  end,
})
```

**Reply by andreacfromtheapp** [reply]
this works better but something weird still happens: it works fine the first time, then when trying projects, session, recent from the dashboard, it is stuck on the dashboard. it was doing this with the first version as well, but I thought it was my messing around fault.

**Reply by dpetka2001** [reply]
I don't understand what you're saying. Are you invoking `:Dashboard` again when you are inside a Neovim session?

**Reply by andreacfromtheapp** [reply]
> I don't understand what you're saying. Are you invoking `:Dashboard` again when you are inside a Neovim session?

I launch nvim, then I try to invoke either Projects or Restore Session or Recents Files but nothing happens. It stays on the dashboard. Or if it opens, it opens the tabs _in background/behind the dashboard_  weird.

**Reply by dpetka2001** [reply]
Maybe try adding `BufReadPost` to the events. Look in `:h events` what events are available, read what they do and try to decide which ones fit the best your needs.

**Reply by andreacfromtheapp** [reply]
@dpetka2001 I tried playing around with events and still get weird behavior. Not sure why nor skilled enough to troubleshoot. At this  point I'm kinda giving up since it's not that crucial. It would have been nice but cba.
